### PR TITLE
Sas UI d2 1844 new cstg key pair

### PR DESCRIPTION
**Pull request by ssundahlTTD**
Allow the user to create a new CSTG key pair.  Add the creation dialog to the key pair screen.

image

image

### CODE REVIEW

File: src/api/routers/participantsRouter.ts [hunk 1]
```diff
@@ -243,6 +249,44 @@ export function createParticipantsRouter() {
...
+      );
+
+      const keyPairs = await addKeyPair(participant.siteId, name, disabled);
+
+      await updateAuditTrailToProceed(auditTrail.id);
+      return res.status(201).json(keyPairs);
+    }
+  );
+
+  participantsRouter.get(
```
Comment (jingyi-gao-ttd) [inline src/api/routers/participantsRouter.ts:277]: nit: May I ask for the reason for moving this from site routes? For me, I feel as the participant route keeps growing, it makes sense to have those site-oriented APIs to be part of sites routes? (Key pair management, sharing permission management and domain management)

File: src/web/components/KeyPairs/KeyPairDialog.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import { ComponentMeta, ComponentStory } from '@storybook/react';
```
Comment (jingyi-gao-ttd) [inline src/web/components/KeyPairs/KeyPairDialog.stories.tsx:1]: nit: for the new story could we use CSF 3.0

here is the doc: https://storybook.js.org/docs/react/writing-stories/introduction

File: src/web/components/KeyPairs/KeyPairDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    >
+      <Form<AddKeyPairFormProps>
+        onSubmit={onSubmit}
+        submitButtonText='Create Key Pair'
+        defaultValues={keyPair}
+      >
+        <TextInput inputName='name' label='Name' />
+        <CheckboxInput
+          inputName='disabled'
+          label='Disabled'
```
Comment (jingyi-gao-ttd) [inline src/web/components/KeyPairs/KeyPairDialog.tsx:44]: I think it looks a bit off to have Disabled and checkbox in two lines. could we try
```
<CheckboxInput
          inputName='disabled'
          options={[{ optionLabel: 'Disabled', value: false }]}
        />
```

File: src/api/routers/participantsRouter.ts (hunk 1 above)
Comment (ssundahlTTD) [reply src/api/routers/participantsRouter.ts:277]: The main reason is that I wanted this behind participant validation and didn't want to recreate that in another router.  The way I wrote this, the participant will only see their own sites (once that's done on the back end).  As it was implemented in the site router, anyone could craft a request to see another site's keys without any sort of permission checks.

File: src/web/components/KeyPairs/KeyPairDialog.tsx (hunk 3 above)
Comment (ssundahlTTD) [reply src/web/components/KeyPairs/KeyPairDialog.tsx:44]: Good idea.  I changed it.

File: src/web/components/KeyPairs/KeyPairDialog.stories.tsx (hunk 2 above)
Comment (ssundahlTTD) [reply src/web/components/KeyPairs/KeyPairDialog.stories.tsx]: I changed it.

**Pull request merged**